### ISSUE TITLE
[Bug]: TRT backend crashes upon request cancellation whilst streaming and logprobs are enabled

### ISSUE THREAD

**Issue by shadeMe** [NVIDIA/TensorRT-LLM#10810]
### System Info

The TRT `LLM` backend consistently crashes under the following conditions:
- Logits are gathered for logprob calculation (`gather_generation_logits=True`).
- At least the generated token's logprob is requested in the sampling parameters (`SamplingParams.logprobs=1`).
- The `LLM.generate_async` method is used to initiate generation in a streaming manner (`streaming=True`).
- The `RequestOutput` yielded by the generator is cancelled using the `abort` method.
  - The cancellation needs to happen when the request is still in-flight, i.e., it's scheduled on the active executor and has not completed.

### Environment
- CPU architecture: x86_64
- CPU/Host memory size (if known): 128GiB
- GPU properties
  - GPU name: L4
  - GPU memory size: 24GiB
- Libraries
  - TensorRT-LLM branch or tag: 1.1.0 and 1.2.0.rc7
  - Container used: Built bundled Docker image + Full Build w/t C++ compilation locally
  - Also tested with official wheels on PyPi and Nvidia PyPi in fresh venvs
- NVIDIA driver version: Driver Version: 580.105.08
- CUDA Version: 13.0
- OS: Ubuntu 24.04
- Any other information that may be useful in reproducing the bug

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```python
from tensorrt_llm import SamplingParams
from tensorrt_llm._tensorrt_engine import LLM
import asyncio
import time

async def main():
    llm = LLM(
        model="HuggingFaceTB/SmolLM2-1.7B-Instruct",
        gather_generation_logits=True,
    )

    sampling_params = SamplingParams(
        temperature=0.8,
        top_p=0.95,
        max_tokens=1000,
        logprobs=1,
    )

    async for output in llm.generate_async(
        "Using emojis, CJK and UTF-8 characters, expand this sentence into a 500 word paragraph: Hello 🌍 世界 programming! 深度学习",
        sampling_params,
        streaming=True,
    ):
        # Bug is triggered when cancelling an in-flight request, either
        # immediately or after some steps.
        output.abort()
        print("Aborted generation")
        break

    for i in range(25):
        print(f"Waiting {i}...")
        time.sleep(0.1)

    print("Final output: ", output.outputs[0].text)

if __name__ == "__main__":
    asyncio.run(main())

```

### Expected behavior

Shouldn't crash; request cancellation should gracefully abort in-flight requests.

### actual behavior

Crashes.

## Output
```
/home/user/.pyenv/versions/3.12.12/envs/trt-312/lib/python3.12/site-packages/torch/cuda/__init__.py:63: FutureWarning: The pynvml package is deprecated. Please install nvidia-ml-py instead. If you did not install pynvml directly, please report this to the maintainers of the package that installed pynvml for you.
  import pynvml  # type: ignore[import]
Skipping import of cpp extensions due to incompatible torch version 2.9.0+cu130 for torchao version 0.15.0             Please see https://github.com/pytorch/ao/issues/2919 for more info
[TensorRT-LLM] TensorRT LLM version: 1.1.0
[01/19/2026-11:35:28] [TRT-LLM] [I] Using LLM with TensorRT backend
[01/19/2026-11:35:28] [TRT-LLM] [W] Using default gpus_per_node: 1
[01/19/2026-11:35:28] [TRT-LLM] [I] Set nccl_plugin to None.
Loading Model: [1/3]    Downloading HF model
Downloaded model to /home/user/.cache/huggingface/hub/models--HuggingFaceTB--SmolLM2-1.7B-Instruct/snapshots/31b70e2e869a7173562077fd711b654946d38674
Time: 0.404s
Loading Model: [2/3]    Loading HF model to memory
`torch_dtype` is deprecated! Use `dtype` instead!
174it [00:00, 2648.51it/s]
Time: 0.383s
Loading Model: [3/3]    Building TRT-LLM engine
Time: 42.376s
Loading model done.
Total latency: 43.163s
The argument `trust_remote_code` is to be used with Auto classes. It has no effect here and is ignored.
[TensorRT-LLM][INFO] Engine version 1.1.0 found in the config file, assuming engine(s) built by new builder API.
[TensorRT-LLM][INFO] Refreshed the MPI local session
[TensorRT-LLM][INFO] MPI size: 1, MPI local size: 1, rank: 0
[TensorRT-LLM][INFO] Using user-specified devices: (0)
[TensorRT-LLM][INFO] Rank 0 is using GPU 0
[TensorRT-LLM][INFO] TRTGptModel maxNumSequences: 2048
[TensorRT-LLM][INFO] TRTGptModel maxBatchSize: 2048
[TensorRT-LLM][INFO] TRTGptModel maxBeamWidth: 1
[TensorRT-LLM][INFO] TRTGptModel maxSequenceLen: 8192
[TensorRT-LLM][INFO] TRTGptModel maxDraftLen: 0
[TensorRT-LLM][INFO] TRTGptModel mMaxAttentionWindowSize: (8192) * 24
[TensorRT-LLM][INFO] TRTGptModel enableTrtOverlap: 0
[TensorRT-LLM][INFO] TRTGptModel normalizeLogProbs: 0
[TensorRT-LLM][INFO] TRTGptModel maxNumTokens: 8192
[TensorRT-LLM][INFO] TRTGptModel maxInputLen: 8191 = min(maxSequenceLen - 1, maxNumTokens) since context FMHA and usePackedInput are enabled
[TensorRT-LLM][INFO] TRTGptModel If model type is encoder, maxInputLen would be reset in trtEncoderModel to maxInputLen: min(maxSequenceLen, maxNumTokens).
[TensorRT-LLM][INFO] Capacity Scheduler Policy: GUARANTEED_NO_EVICT
[TensorRT-LLM][INFO] Context Chunking Scheduler Policy: None
[TensorRT-LLM][INFO] Loaded engine size: 3280 MiB
[TensorRT-LLM][INFO] Engine load time 3034 ms
[TensorRT-LLM][INFO] Inspecting the engine to identify potential runtime issues...
[TensorRT-LLM][INFO] The profiling verbosity of the engine does not allow this analysis to proceed. Re-build the engine with 'detailed' profiling verbosity to get more diagnostics.
[TensorRT-LLM][INFO] [MemUsageChange] Allocated 768.01 MiB for execution context memory.
[TensorRT-LLM][INFO] gatherContextLogits: 0
[TensorRT-LLM][INFO] gatherGenerationLogits: 1
[TensorRT-LLM][INFO] [MemUsageChange] TensorRT-managed allocation in IExecutionContext creation: CPU +0, GPU +0, now: CPU 0, GPU 3266 (MiB)
[TensorRT-LLM][INFO] [MemUsageChange] Allocated 3.19 GB GPU memory for runtime buffers.
[TensorRT-LLM][INFO] [MemUsageChange] Allocated 1.88 GB GPU memory for decoder.
[TensorRT-LLM][INFO] Memory usage when calculating max tokens in paged kv cache: total: 22.03 GiB, available: 12.78 GiB
[TensorRT-LLM][INFO] Blocks per window size:
[TensorRT-LLM][INFO] [windowSize=8192] {.primaryBlocks=1963, .secondayBlocks=0}
[TensorRT-LLM][INFO] Max KV cache blocks per sequence: 256 [window size=8192], tokens per block=32, primary blocks=1963, secondary blocks=0
[TensorRT-LLM][INFO] Number of tokens per block: 32.
[TensorRT-LLM][INFO] [MemUsageChange] Allocated 11.50 GiB for max tokens in paged KV cache (62816).
[TensorRT-LLM][INFO] CacheTransceiver is disabled.
Aborted generation
Waiting 0...
terminate called after throwing an instance of 'c10::Error'
  what():  Trying to create tensor with negative dimension -4: [-4, 1, 49152]
Exception raised from check_size_nonnegative at /pytorch/aten/src/ATen/EmptyTensor.h:8 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x80 (0x7c122237cb80 in /home/user/.pyenv/versions/3.12.12/envs/trt-312/lib/python3.12/site-packages/torch/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 0x65 (0x7c122230efbf in /home/user/.pyenv/versions/3.12.12/envs/trt-312/lib/python3.12/site-packages/torch/lib/libc10.so)
frame #2: at::TensorMaker::make_tensor() + 0x3ee (0x7c1200aea02e in /home/user/.pyenv/versions/3.12.12/envs/trt-312/lib/python3.12/site-packages/torch/lib/libtorch_cpu.so)
frame #3: <unknown function> + 0x35ce04 (0x7c105b55ce04 in /home/user/.pyenv/versions/3.12.12/envs/trt-312/lib/python3.12/site-packages/tensorrt_llm/bindings.cpython-312-x86_64-linux-gnu.so)
frame #4: <unknown function> + 0x3cb8b4 (0x7c105b5cb8b4 in /home/user/.pyenv/versions/3.12.12/envs/trt-312/lib/python3.12/site-packages/tensorrt_llm/bindings.cpython-312-x86_64-linux-gnu.so)
frame #5: <unknown function> + 0x3cb924 (0x7c105b5cb924 in /home/user/.pyenv/versions/3.12.12/envs/trt-312/lib/python3.12/site-packages/tensorrt_llm/bindings.cpython-312-x86_64-linux-gnu.so)
frame #6: <unknown function> + 0x3cb96c (0x7c105b5cb96c in /home/user/.pyenv/versions/3.12.12/envs/trt-312/lib/python3.12/site-packages/tensorrt_llm/bindings.cpython-312-x86_64-linux-gnu.so)
frame #7: <unknown function> + 0x427b42 (0x7c105b627b42 in /home/user/.pyenv/versions/3.12.12/envs/trt-312/lib/python3.12/site-packages/tensorrt_llm/bindings.cpython-312-x86_64-linux-gnu.so)
<omitting python frames>
frame #22: <unknown function> + 0x9caa4 (0x7c1222e9caa4 in /lib/x86_64-linux-gnu/libc.so.6)
frame #23: <unknown function> + 0x129c6c (0x7c1222f29c6c in /lib/x86_64-linux-gnu/libc.so.6)

[hostname:1782217] *** Process received signal ***
[hostname:1782217] Signal: Aborted (6)
[hostname:1782217] Signal code:  (-6)
[hostname:1782217] [ 0] /lib/x86_64-linux-gnu/libc.so.6(+0x45330)[0x7c1222e45330]
[hostname:1782217] [ 1] /lib/x86_64-linux-gnu/libc.so.6(pthread_kill+0x11c)[0x7c1222e9eb2c]
[hostname:1782217] [ 2] /lib/x86_64-linux-gnu/libc.so.6(gsignal+0x1e)[0x7c1222e4527e]
[hostname:1782217] [ 3] /lib/x86_64-linux-gnu/libc.so.6(abort+0xdf)[0x7c1222e288ff]
[hostname:1782217] [ 4] /lib/x86_64-linux-gnu/libstdc++.so.6(+0xa5ff5)[0x7c121a8a5ff5]
[hostname:1782217] [ 5] /lib/x86_64-linux-gnu/libstdc++.so.6(+0xbb0da)[0x7c121a8bb0da]
[hostname:1782217] [ 6] /lib/x86_64-linux-gnu/libstdc++.so.6(__cxa_call_terminate+0x33)[0x7c121a8a58e6]
[hostname:1782217] [ 7] /lib/x86_64-linux-gnu/libstdc++.so.6(__gxx_personality_v0+0x31a)[0x7c121a8ba8ba]
[hostname:1782217] [ 8] /lib/x86_64-linux-gnu/libgcc_s.so.1(+0x22b06)[0x7c122282bb06]
[hostname:1782217] [ 9] /lib/x86_64-linux-gnu/libgcc_s.so.1(_Unwind_Resume+0x12d)[0x7c122282c5cd]
[hostname:1782217] [10] /home/user/.pyenv/versions/3.12.12/envs/trt-312/lib/python3.12/site-packages/tensorrt_llm/bindings.cpython-312-x86_64-linux-gnu.so(+0x3cb900)[0x7c105b5cb900]
[hostname:1782217] [11] /home/user/.pyenv/versions/3.12.12/envs/trt-312/lib/python3.12/site-packages/tensorrt_llm/bindings.cpython-312-x86_64-linux-gnu.so(+0x3cb924)[0x7c105b5cb924]
[hostname:1782217] [12] /home/user/.pyenv/versions/3.12.12/envs/trt-312/lib/python3.12/site-packages/tensorrt_llm/bindings.cpython-312-x86_64-linux-gnu.so(+0x3cb96c)[0x7c105b5cb96c]
[hostname:1782217] [13] /home/user/.pyenv/versions/3.12.12/envs/trt-312/lib/python3.12/site-packages/tensorrt_llm/bindings.cpython-312-x86_64-linux-gnu.so(+0x427b42)[0x7c105b627b42]
[hostname:1782217] [14] /home/user/.pyenv/versions/3.12.12/lib/libpython3.12.so.1.0(PyObject_CallOneArg+0x65)[0x7c1223379dc5]
[hostname:1782217] [15] /home/user/.pyenv/versions/3.12.12/lib/libpython3.12.so.1.0(_PyObject_GenericGetAttrWithDict+0x96)[0x7c12233dc636]
[hostname:1782217] [16] /home/user/.pyenv/versions/3.12.12/lib/libpython3.12.so.1.0(PyObject_GetAttr+0x38)[0x7c12233dbd28]
[hostname:1782217] [17] /home/user/.pyenv/versions/3.12.12/lib/libpython3.12.so.1.0(_PyEval_EvalFrameDefault+0x1694)[0x7c1223314264]
[hostname:1782217] [18] /home/user/.pyenv/versions/3.12.12/lib/libpython3.12.so.1.0(_PyObject_FastCallDictTstate+0x69)[0x7c122337bad9]
[hostname:1782217] [19] /home/user/.pyenv/versions/3.12.12/lib/libpython3.12.so.1.0(_PyObject_Call_Prepend+0x110)[0x7c122337bda0]
[hostname:1782217] [20] /home/user/.pyenv/versions/3.12.12/lib/libpython3.12.so.1.0(+0x1f9f0d)[0x7c12233f9f0d]
[hostname:1782217] [21] /home/user/.pyenv/versions/3.12.12/lib/libpython3.12.so.1.0(_PyObject_MakeTpCall+0x80)[0x7c1223379bd0]
[hostname:1782217] [22] /home/user/.pyenv/versions/3.12.12/lib/libpython3.12.so.1.0(_PyEval_EvalFrameDefault+0x439f)[0x7c1223316f6f]
[hostname:1782217] [23] /home/user/.pyenv/versions/3.12.12/lib/libpython3.12.so.1.0(+0x17d733)[0x7c122337d733]
[hostname:1782217] [24] /home/user/.pyenv/versions/3.12.12/lib/libpython3.12.so.1.0(_PyEval_EvalFrameDefault+0x87d)[0x7c122331344d]
[hostname:1782217] [25] /home/user/.pyenv/versions/3.12.12/lib/libpython3.12.so.1.0(+0x17d733)[0x7c122337d733]
[hostname:1782217] [26] /home/user/.pyenv/versions/3.12.12/lib/libpython3.12.so.1.0(+0x385eba)[0x7c1223585eba]
[hostname:1782217] [27] /home/user/.pyenv/versions/3.12.12/lib/libpython3.12.so.1.0(+0x3078fb)[0x7c12235078fb]
[hostname:1782217] [28] /lib/x86_64-linux-gnu/libc.so.6(+0x9caa4)[0x7c1222e9caa4]
[hostname:1782217] [29] /lib/x86_64-linux-gnu/libc.so.6(+0x129c6c)[0x7c1222f29c6c]
[hostname:1782217] *** End of error message ***
[1]    1782217 IOT instruction (core dumped)  /home/user/.pyenv/versions/3.12.12/envs/trt-312/bin/python 
```

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and checked the [documentation](https://nvidia.github.io/TensorRT-LLM/) and [examples](https://github.com/NVIDIA/TensorRT-LLM/tree/main/examples) for answers to frequently asked questions.

**Comment by karljang** [comment]
@shadeMe, thank you for bringing this issue to our attention.

The issue has been forwarded to the team. However, I’m curious if you’d be able to try using the PyTorch workflow instead?

Since the 1.0 release, TensorRT LLM has transitioned to the PyTorch workflow. The team is currently prioritizing the PyTorch backend, so support for the TensorRT backend will be limited.

**Comment by shadeMe** [comment]
> [@shadeMe](https://github.com/shadeMe), thank you for bringing this issue to our attention.
> 
> The issue has been forwarded to the team. However, I’m curious if you’d be able to try using the PyTorch workflow instead?
> 
> Since the 1.0 release, TensorRT LLM has transitioned to the PyTorch workflow. The team is currently prioritizing the PyTorch backend, so support for the TensorRT backend will be limited.

Thanks! We discovered this issue before we realized that the TRT backend was deprecated. We intend to use the PyTorch backend going forward (it doesn't exhibit this issue either) but wanted to flag it all the same.
